### PR TITLE
[lift] LiftStorage wrapper

### PR DESCRIPTION
**Pull request by kvark**
Based on #104

### CODE REVIEW

File: rspirv/lift/storage.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use std::{
+    borrow::Borrow,
+    collections::hash_map::{Entry, VacantEntry, HashMap},
+    hash::BuildHasherDefault,
+};
+
+use fxhash::FxHasher;
+
+
+pub struct LiftStorage<T, L = Token<T>> {
```
Comment (antiagainst) [inline rspirv/lift/storage.rs]: Can we please have documentation for public APIs?

File: rspirv/lift/storage.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    ) -> (Token<T>, VacantEntry<spirv::Word, L>) {
+        let token = self.values.append(value);
+        match self.lookup.entry(id) {
+            Entry::Occupied(_) => panic!("Id {:?} is already used", id),
+            Entry::Vacant(e) => (token, e),
+        }
+    }
+}
+
+impl<T> LiftStorage<T> {
```
Comment (antiagainst) [inline rspirv/lift/storage.rs]: It has been a while and this syntax is not quite familiar to me anymore. What's the difference of this `impl` compared to the above one?

File: rspirv/lift/storage.rs (hunk 1 above)
Comment (kvark) [reply rspirv/lift/storage.rs]: yes, sure 😅

File: rspirv/lift/storage.rs (hunk 2 above)
Comment (kvark) [reply rspirv/lift/storage.rs]: The one above is generic across `L`. This block omits `L` thus it's for the default `L = Token<T>` only, which is a narrower subset of all.

**Pull request merged**